src/ProjectForm.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from 'react';
+import { Card, CardContent, Stack, TextField, Button } from '@mui/material';
+import Paper from '@mui/material/Paper';
+import { styled } from '@mui/material/styles';
+import { DesktopDatePicker, LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
+import moment from 'moment';
+
+const Item = styled(Paper)(({ theme }) => ({
+  backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
+  ...theme.typography.body2,
+  padding: theme.spacing(1),
+  color: theme.palette.text.secondary,
+  alignItems: 'center',
+  width: '100%',
+}));
+
+function ProjectForm() {
+  const [name, setName] = useState('');
+  const [projectName, setProjectName] = useState('');
+  const [date, setDate] = useState(moment());
+  const [description, setDescription] = useState('');
+
+  function handleChange(e) {
+    switch (e.target.id) {
+      case 'name':
+        setName(e.target.value);
+
+        break;
+      case 'description':
+        setDescription(e.target.value);
+        break;
+    }
+  }
+
+  function handleDateChange(date) {
+    setDate(date);
+  }
+
+  useEffect(() => {
+    setProjectName(name.replace(/[^A-Za-z_0-9]/g, '-'));
+  }, [name]);
+
+  return (
+    <header className="preview">
+      <Card>
+        <CardContent>
+          <Stack direction="column" justifyContent="center" alignItems="center" spacing={2}>
+            <h2>Project Information</h2>
+            <Item>
+              <TextField
+                id="name"
+                label="Name"
+                inputProps={{ maxLength: 100 }}
+                onChange={handleChange}
+                value={name}
+                helperText={`Your project name will be ${projectName}`}
+                fullWidth
+              />
+            </Item>
+            <Item>
+              <LocalizationProvider dateAdapter={AdapterMoment}>
+                <DesktopDatePicker
+                  label="Date"
+                  inputFormat="MM/DD/YYYY"
```

Comment:
in one of the auth worker functions, we are generating a random name for the `repo`  using the `en-CA` locale in the use case where the user does not enter a value for the repo name. I would make the date format consistent with what we already have.